src/OSPSuite.SimModelNative/src/DESolver.cpp

```diff
@@ -440,11 +440,16 @@ namespace SimModelNative
 			if (!observer->IsUsedInSimulation())
 				continue;
 
-			//calculate oberver threshold as f(ODEVariable_Thresholds, 0.0)
-			//(where f(y,t) is the calculation formula of the observer
-			double observerThreshold= observer->CalculateValue(odeVariableThresholds, 0.0, USE_SCALEFACTOR);
-
-			observer->SetComparisonThreshold(observerThreshold);
+         if(observer->IsConstantDuringCalculation())
+            observer->SetComparisonThreshold(variableThreshold);
+         else
+         {
+            //calculate observer threshold as f(ODEVariable_Thresholds, 0.0)
+            //(where f(y,t) is the calculation formula of the observer
+            double observerThreshold = observer->CalculateValue(odeVariableThresholds, 0.0, USE_SCALEFACTOR);
```

Comment:
The initial idea was to set the threshold just by calculating the observer formula by using the thresholds as variable values.
Thus if observer is defined as e.g. `alpha*Variable` then the observer threshold would be calculated as `alpha*Variable_Threshold` etc.
PArticularly e.g. for concentration observers it would be calculated as `Amount_Threshold/Volume`.
But if the amount is constant, observer is just replaced by the constant value and threshold calculation returns just this value instead.
So now I just check for constant observers explicitly and set the threshold to the absolute tolerance value